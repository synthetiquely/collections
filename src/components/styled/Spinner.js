import React from 'react';
import glamorous from 'glamorous';
import { css } from 'glamor';

const animationStyles = () => {
  const spin = css.keyframes({
    '0%': { transform: 'rotate(0deg)' },
    '100%': { transform: 'rotate(360deg)' },
  });
  return {
    display: 'block',
    width: '60px',
    height: '60px',
    margin: '5px',
    border: '5px solid #ffdb4d',
    borderTopColor: 'transparent',
    borderRadius: '50%',
    animation: `${spin} 1s linear infinite`,
  };
};

const Overlay = glamorous.div({
  position: 'fixed',
  top: '0',
  right: '0',
  bottom: '0',
  left: '0',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  backgroundColor: '#fff',
});

const Spinner = glamorous.span(animationStyles);

export default ({ overlayed, ...rest }) => {
  if (overlayed) {
    return (
      <Overlay>
        <Spinner {...rest} />
      </Overlay>
    );
  }
  return <Spinner {...rest} />;
};
