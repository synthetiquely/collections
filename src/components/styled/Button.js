import React from 'react';
import glamorous from 'glamorous';
import Spinner from './Spinner';

const Button = glamorous.button(
  {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    fontSize: '16px',
    padding: '10px',
    color: '#000',
    border: '1px solid #ccc',
    outline: 'none',
    backgroundColor: '#fff',
    cursor: 'pointer',
    ':hover,:active,:focus': {
      borderLeft: '1px solid',
      borderColor: '#ffdb4d',
    },
  },
  ({ size, inverted, bordered }) => {
    let styles = {};
    if (size === 'fullWidth') {
      styles = {
        ...styles,
        width: '100%',
      };
      if (inverted) {
        styles = {
          ...styles,
          ':hover,:active,:focus': {
            backgroundColor: '#ffdb4d',
            color: '#fff',
            borderColor: '#ffd35f',
          },
        };
      }
      if (bordered) {
        styles = {
          ...styles,
          borderRadius: '10px',
        };
      }
    }
    return styles;
  },
);

export default ({ loading, ...rest }) => (
  <Button {...rest}>
    {loading && <Spinner size="sm" />}{' '}
    <span style={{ padding: '0 15px' }}>{rest.children}</span>
  </Button>
);
