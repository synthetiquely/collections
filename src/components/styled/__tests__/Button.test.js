import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Button from '../Button';
import Spinner from '../Spinner';

describe('Button Component', () => {
  it('should render and match its own snapshot', () => {
    const wrapper = mount(<Button />);
    const tree = renderer.create(<Button />).toJSON();

    expect(wrapper).toHaveLength(1);
    expect(tree).toMatchSnapshot();
  });

  it('should render text passed as children to the component', () => {
    const wrapper = mount(<Button>test text</Button>);

    expect(wrapper.find('button').text()).toEqual(' test text');
  });

  it('should render a spinner, if loading is true', () => {
    const wrapper = mount(<Button loading>test text</Button>);

    expect(wrapper.find(Spinner)).toHaveLength(1);
  });
});
