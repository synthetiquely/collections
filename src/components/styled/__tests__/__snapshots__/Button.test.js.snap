// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render and match its own snapshot 1`] = `
<button
  className="css-g7cb4b"
>
   
  <span
    style={
      Object {
        "padding": "0 15px",
      }
    }
  />
</button>
`;
